### PR TITLE
Users/tivu/webpack

### PR DESCRIPTION
**Pull request by TienSFU25**
Just dumping up here so you can see what I'm doing. Doesn't run properly yet

### CODE REVIEW

**Comment by TienSFU25** [comment]
This PR is to introduce Webpack with Babel and React into the project.

I did not intend this PR to be so big. Originally I only wanted to change the "stimulus" from "external-html-keyboard" to "html-keyboard", using ReactDOMServer.renderToStaticMarkup(). But this required a chain of changes required to structurally enable React to be used in the project.

Important bits in this change:
- webpack.config.js
- usage of es6 style "import" and "export"s for modules
- React components

Going to omit Typescript and Redux for now I think, until we find it truly necessary. I think this is enough structure.

File: public/scripts/experiments/jnd.js [hunk 1]
```diff
@@ -524,178 +546,4 @@ class JND {
     hiddenElement.download = this.condition_name + "_jnd_summary_results.csv";
     hiddenElement.click();
   }
-
-
-  /**
-   * Performs the necessary D3 operations to plot distributions depending on graph type.
-   */
-  plot_distributions() {
```
Comment (coyiutoc) [inline public/scripts/experiments/jnd.js:535]: @TienSFU25 Did you end up relocating all the plotting code somewhere? Can't find it in your changes anymore?

Comment (TienSFU25) [reply public/scripts/experiments/jnd.js:535]: its in JNDTrialDisplay.jsx now

File: public/components/Index/index.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import JND from '../JND';
+import DNJ from '../DNJ';
```
Comment (coyiutoc) [inline public/components/Index/index.jsx]: Can you change all your DNJs to Stevens? I think you meant Stevens...

File: public/scripts/experiments/jnd.js [hunk 3]
```diff
@@ -127,19 +140,19 @@ class JND {
...
     // Initialize a variable for this so it is usable inside on_start
-    var jnd_exp = this; 
+    var jnd_exp = this;
 
     var trial = {
-      type:'external-html-keyboard-response',
-      url: localhost + "/jnd_trial",
+      type:'html-keyboard-response',
+      // url: LOCAL_HOST + "/jnd_trial",
+      stimulus: "<div>UBC, I love you!</div>",
```
Comment (coyiutoc) [inline public/scripts/experiments/jnd.js:119]: Is this needed? Hahah

Comment (TienSFU25) [reply public/scripts/experiments/jnd.js:119]: an opportunity to express one's affection!

**Pull request merged**